src/main/java/programminglife/model/drawing/DrawableNode.java

```diff
@@ -1,198 +1,134 @@
 package programminglife.model.drawing;
 
-import javafx.scene.paint.Color;
 import javafx.scene.shape.Rectangle;
-import programminglife.model.*;
+import programminglife.model.GenomeGraph;
+import programminglife.model.Link;
+import programminglife.model.XYCoordinate;
 
 import java.util.Collection;
-import java.util.LinkedHashSet;
-import java.util.NoSuchElementException;
 
 /**
- * A {@link Segment} that also Implements {@link Drawable}.
+ * Created by toinehartman on 12/06/2017.
  */
-public class DrawableNode extends Rectangle {
-    private Node node;
-    private boolean drawDimensionsUpToDate = false;
-    private Collection<Node> parents;
-    private Collection<Node> children;
+public abstract class DrawableNode extends Rectangle implements Drawable {
+    private final GenomeGraph graph;
+    private final int id;
 
+    private boolean drawDimensionsUpToDate = false;
 
     /**
-     * Create a DrawableSegment from a Segment.
-     * @param node The segment to create this DrawableSegment from.
+     * Construct a {@link DrawableNode}.
+     * @param graph the {@link GenomeGraph} to draw from
+     * @param id the ID of the {@link DrawableNode}
      */
-    public DrawableNode(Node node) {
-        this.node = node;
-        parents = new LinkedHashSet<>(node.getParents());
-        children = new LinkedHashSet<>(node.getChildren());
-        this.setDrawDimensions();
+    DrawableNode(GenomeGraph graph, int id) {
+        this.graph = graph;
+        this.id = id;
     }
 
     /**
-     * Get all the children of the node {@link DrawableNode}.
-     * @return children {@link Collection} are the direct children of the node {@link DrawableNode}.
+     * Get the ID.
+     * @return the ID
      */
-    public Collection<Node> getChildren() {
-        return children;
+    public final int getIdentifier() {
+        return this.id;
     }
 
     /**
-     * Get all the parents of the node {@link DrawableNode}.
-     * @return parent {@link Collection} are the direct parents of the node {@link DrawableNode}.
-     **/
-    Collection<Node> getParents() {
-        return parents;
+     * Get the {@link GenomeGraph}.
+     * @return the graph
+     */
+    public final GenomeGraph getGraph() {
+        return graph;
     }
 
     /**
-     * Replace a child node with a dummy node.
-     * @param oldChild The {@link DrawableNode} to replace.
-     * @param newChild The {@link DrawableNode} to replace with.
+     * Set if the dimensions are up to date.
+     * @param upToDate if is is
```

Comment:
I think this javadoc is a bit unclear. 

I would suggest: @param upToDate {@link boolean} true if up to date else false.